newIDE/app/src/GameDashboard/GameAnalyticsEvaluator.js

```diff
@@ -0,0 +1,650 @@
+          b.players.d0PlayersBelow900s != null
+            ? a.players.d0PlayersBelow900s + b.players.d0PlayersBelow900s
+            : undefined,
+      },
+
+    retention: null,
+  };
+};
+
+/**
+ * @param gameMetrics today first
+ * @returns metrics summed by weeks
+ * @see mergeGameMetrics
+ */
+const mergeGameMetricsByWeek = (
+  gameMetrics: GameMetrics[]
+): MergedGameMetrics[] => {
+  const mergedGameMetrics: Array<MergedGameMetrics> = [];
+  for (let weekIndex = 0; weekIndex < gameMetrics.length; weekIndex += 7) {
+    let mergedGameMetric = gameMetrics[weekIndex];
+    for (
+      let index = weekIndex + 1;
+      index < weekIndex + 7 && index < gameMetrics.length;
+      index++
+    ) {
+      mergedGameMetric = mergeGameMetrics(mergedGameMetric, gameMetrics[index]);
+    }
+    mergedGameMetrics.push(((mergedGameMetric: any): MergedGameMetrics));
+  }
+  return mergedGameMetrics;
+};
+
+/**
+ * @param playersBelowSums
+ * @param playersCount
+ * @returns the duration limit that split the players the most evenly.
+ * The real median would hard to process for the backend,
+ * so we rely on this.
+ */
+const findNearestToMedianDurationIndex = (
+  playersBelowSums: Array<number>,
+  playersCount: number
+): number => {
+  const overMedianDurationIndex = playersBelowSums.findIndex(
+    (value, index) => index > 0 && value > playersCount / 2
+  );
+  const overMedianDuration = playersBelowSums[overMedianDurationIndex];
+  const underMedianDuration = playersBelowSums[overMedianDurationIndex + 1];
+  return underMedianDuration &&
+    Math.abs(overMedianDuration - playersCount / 2) <
+      Math.abs(underMedianDuration - playersCount / 2)
+    ? overMedianDurationIndex
+    : overMedianDurationIndex + 1;
+};
+
+/**
+ * @param playersBelowSums
+ * @param viewersCount
+ * @returns the duration limit which maximize: duration * players.
+ * It allows to give the most impressive point from `overPlayedDuration`.
+ */
+const findGreaterDurationPlayerIndex = (
+  playersBelowSums: Array<number>,
+  viewersCount: number
+): number => {
+  let durationPlayerMax = 0;
+  let greaterDurationPlayerIndex = 0;
+  for (let index = 0; index < playersBelowSums.length; index++) {
+    const playersOverSum = viewersCount - playersBelowSums[index];
+    const duration = durationValues[index];
+
+    const durationPlayer = playersOverSum * duration;
+    if (durationPlayer >= durationPlayerMax) {
+      durationPlayerMax = durationPlayer;
+      greaterDurationPlayerIndex = index;
+    }
+  }
+  return greaterDurationPlayerIndex;
+};
+
+const evaluatePlayersOverDurationPercent = (
+  playersBelowDuration: ?number,
+  playersCount: number
+): number => {
+  return playersBelowDuration != null && playersCount > 0
+    ? (100 * (playersCount - playersBelowDuration)) / playersCount
+    : 0;
+};
+
+const evaluatePlayersBetweenDurationPercent = (
+  playersBelowUpperDuration: ?number,
+  playersBelowLowerDuration: ?number,
+  playersCount: number
+): number => {
```

Comment:
I chose this order because it's the subtraction operand order, but maybe it should be the other way around?

Reply:
If you're worried the function would be misused, then maybe you can use an object with names keys as a param, rather than 3 params.
I think it's fine though, as you use it only in one place